### PR TITLE
Map tests

### PR DESCRIPTION
**Pull request by ashleyjelks**
Enables/updates tests in the Global Map component.

#trivial

### CODE REVIEW

File: src/lib/Scenes/Map/GlobalMap.tsx [hunk 1]
```diff
@@ -604,6 +604,7 @@ export class GlobalMap extends React.Component<Props, State> {
           resizeMode="cover"
           style={{ ...this.backgroundImageSize }}
         />
+        {console.log("TCL: render -> this.props", this.props)}
```
Comment (kierangillen) [inline src/lib/Scenes/Map/GlobalMap.tsx]: Not sure if this log is intentionally left

**Comment by kierangillen** [comment]
Looks good other than the loose log!

File: src/lib/Scenes/Fair/Screens/__tests__/FairBooth-tests.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { FairBoothContainer as FairBooth } from "../FairBooth"
 
 jest.unmock("react-relay")
 
-xit("renders properly", async () => {
+it("renders properly", async () => {
```
Comment (alloy) [inline src/lib/Scenes/Fair/Screens/__tests__/FairBooth-tests.tsx]: This one is really flaky. I disabled it in the RN 0.59 upgrade PR.

File: src/lib/Scenes/Map/GlobalMap.tsx [hunk 3]
```diff
@@ -246,7 +246,7 @@ export class GlobalMap extends React.Component<Props, State> {
       setTimeout(this.resetZoomAndCamera, 500)
     }
 
-    // If there is a new city, emity it and update our map.
+    // If there is a new city, enity it and update our map.
```
Comment (alloy) [inline src/lib/Scenes/Map/GlobalMap.tsx]: ?

```suggestion
    // If there is a new city, emit it and update our map.
```

File: src/lib/Scenes/Map/__tests__/GlobalMap-tests.tsx [hunk 4]
```diff
@@ -27,6 +30,9 @@ describe("GlobalMap", () => {
         }
       `,
       mockData: {
+        safeAreaInsets: {
+          top: 1,
+        },
```
Comment (alloy) [inline src/lib/Scenes/Map/__tests__/GlobalMap-tests.tsx]: This is a regular prop, not GraphQL data. What you want to do is update the MockRelayRenderer and renderRelayTree utils to the latest versions in Reaction and use the component props https://github.com/artsy/reaction/blob/2308594ff1d34da2a089b816413bed6985928445/src/DevTools/MockRelayRenderer.tsx#L25

File: src/lib/__fixtures__/CityFixture.ts [hunk 5]
```diff
@@ -1,15 +1,113 @@
 export const CityFixture = {
-  name: "Paris",
+  safeAreaInsets: {
+    top: 1,
+    bottom: 1,
+    left: 1,
+    right: 1,
+  },
```
Comment (alloy) [inline src/lib/__fixtures__/CityFixture.ts]: As per above, this shouldn't be mixed in with the GraphQL data.

File: src/lib/Scenes/Map/__tests__/GlobalMap-tests.tsx (hunk 4 above)
Comment (ashleyjelks) [reply src/lib/Scenes/Map/__tests__/GlobalMap-tests.tsx]: Super helpful, thanks!

**Comment by ashleyjelks** [comment]
> Do you know what’s up with all the Xcode project changes?

Not sure. These were generated after running `bundle exec pod install` I'll uncommit these file changes.

File: src/lib/Scenes/Map/__tests__/GlobalMap-tests.tsx [hunk 6]
```diff
@@ -1,24 +1,32 @@
...
   setAccessToken: jest.fn(),
+  UserTrackingModes: jest.fn(),
 }))
 
 describe("GlobalMap", () => {
-  xit("renders correctly", async () => {
+  it("renders correctly", async () => {
     await renderRelayTree({
-      Component: GlobalMap,
+      componentProps: {
```
Comment (kierangillen) [inline src/lib/Scenes/Map/__tests__/GlobalMap-tests.tsx]: We'll want to mock the `viewer` object in this `mockData`. Generally, I think the mockData object should match the top level of the corresponding query. So ```mockData: {
        viewer: viewerFixture,
      }```

Comment (ashleyjelks) [reply src/lib/Scenes/Map/__tests__/GlobalMap-tests.tsx]: Not sure I follow what you mean. Would you please elaborate? The `mockData` shape maps this query mockData > viewer > city > cityFixture. 

The props passed to `componentProps` don't go in the mockData object as @alloy pointed out in a previous code review so they are being passed here. The `safeAreaInsets` prop comes Mapbox (not the Graphql data) so it has to be mocked in the `componentProps` field.

Comment (kierangillen) [reply src/lib/Scenes/Map/__tests__/GlobalMap-tests.tsx]: Ah yes you're right! Usually viewer is used when wrapping two or more queries so I thought that was the case here.

**Comment by kierangillen** [comment]
Merged!

**Pull request merged**